source/common/router/router.h

```diff
@@ -67,6 +67,41 @@ class FilterUtility {
     bool hedge_on_per_try_timeout_;
   };
 
+  class StrictHeaderChecker {
+  public:
+    struct HeaderCheckResult {
+      bool valid_ = true;
+      const Http::HeaderEntry* entry_;
+    };
+
+    static const HeaderCheckResult test(Http::HeaderMap& headers,
```

Comment:
Mind adding a comment to this explaining what it does? A more descriptive name might be useful too, maybe `checkHeader`?